weave/trace_server_bindings/async_batch_processor.py

```diff
@@ -96,13 +128,17 @@ def _process_batches(self) -> None:
             if current_batch := self._get_next_batch():
                 try:
                     self.processor_fn(current_batch)
+                    for item in current_batch:
+                        self.queue.task_done()
                 except Exception as e:
                     if get_raise_on_captured_errors():
                         raise
-                    logger.exception(f"Error processing batch: {e}")
-                else:
-                    for _ in current_batch:
-                        self.queue.task_done()
```

Comment:
why did you change this?

Reply:
I probably thought it was more readable, i'm never exactly sure what the try/except/else stuff does. I'll change it back though now I get it